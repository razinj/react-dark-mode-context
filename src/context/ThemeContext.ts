import { createContext } from 'react'

const defaultValue = {
  currentTheme: 'light',
  changeCurrentTheme: (theme: 'light' | 'dark') => {},
}

const ThemeContext = createContext(defaultValue)

export default ThemeContext
