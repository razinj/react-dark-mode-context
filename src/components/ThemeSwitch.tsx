import { FC, useContext } from 'react'
import ThemeContext from '../context/ThemeContext'

const ThemeSwitch: FC = () => {
  const { currentTheme, changeCurrentTheme } = useContext(ThemeContext)

  return (
    <button
      data-testid='switch-theme-btn'
      style={{
        padding: 5,
        borderRadius: 5,
        color: currentTheme === 'light' ? 'white' : 'black',
        background: currentTheme === 'light' ? 'black' : 'white',
      }}
      onClick={() => changeCurrentTheme(currentTheme === 'light' ? 'dark' : 'light')}
    >
      Go {currentTheme === 'light' ? 'DARK MODE' : 'LIGHT MODE'}
    </button>
  )
}

export default ThemeSwitch
