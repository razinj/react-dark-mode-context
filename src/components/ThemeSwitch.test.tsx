import { fireEvent, render } from '@testing-library/react'
import ThemeContext from '../context/ThemeContext'
import ThemeSwitch from './ThemeSwitch'

describe('ThemeSwitch', () => {
  it('should match snapshot', () => {
    const { container } = render(<ThemeSwitch />)

    expect(container).toMatchSnapshot()
  })

  it('should render and call switching theme method correctly', () => {
    const changeCurrentThemeFn = jest.fn()

    const { getByTestId } = render(
      <ThemeContext.Provider
        value={{
          currentTheme: 'light',
          changeCurrentTheme: changeCurrentThemeFn,
        }}
      >
        <ThemeSwitch />
      </ThemeContext.Provider>
    )

    const switchThemeBtn = getByTestId('switch-theme-btn')

    expect(switchThemeBtn).toBeInTheDocument()

    fireEvent.click(switchThemeBtn)

    expect(changeCurrentThemeFn).toBeCalledTimes(1)
    expect(changeCurrentThemeFn).toBeCalledWith('dark')
  })
})
