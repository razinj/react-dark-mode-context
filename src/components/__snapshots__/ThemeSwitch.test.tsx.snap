// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch should match snapshot 1`] = `
<div>
  <button
    data-testid="switch-theme-btn"
    style="padding: 5px; border-radius: 5px; color: white; background: black;"
  >
    Go 
    DARK MODE
  </button>
</div>
`;
